sdk/include/opentelemetry/sdk/trace/span_data.h

```diff
@@ -72,7 +142,7 @@ class SpanData final : public Recordable
    * Get the attributes for this span
    * @return the attributes for this span
    */
-  const std::unordered_map<std::string, common::AttributeValue> &GetAttributes() const noexcept
+  const std::unordered_map<std::string, SpanDataAttributeValue> &GetAttributes() const noexcept
   {
     return attributes_;
```

Comment:
Here we assume that it'd gonna be requiring a copy of the original map passed to SDK. Whereas in a `synchronous` (or `push` , whatever we call it), exporter - that could have been returning a reference to the same container object, as passed to SDK. Basically before SDK API call returns - all contents of the container are still valid, including foreign app-supplied `nostd::string_view` objects. By always copying into our own we incur some penalty, that may be unnecessary for synchronous paths. I'm thinking of use-case where SDK API immediately passes all calls straight to some form of Monitoring Agent, that'd reconcile all data on the other side, on caller / user thread without any async processing, thus not needing a copy.

Reply:
We could've added a class with some form of ownership semantics to original `common::AttributeValue` , and allowed an exporter that has to create a copy - call into `GetAttributes` that returns a reference to original dictionary, then let the exporter do the job of doing a deep copy into their own preferred dictionary type. Basically providing a template to copy rather than implementation of a new value type.